webapi/src/main/scala/org/knora/webapi/slice/admin/domain/model/User.scala

```diff
@@ -136,24 +137,46 @@ final case class User(
   def isAnonymousUser: Boolean = id.equalsIgnoreCase(OntologyConstants.KnoraAdmin.AnonymousUser)
 }
 
-final case class UserIri private (value: String) extends AnyVal
-object UserIri {
-  def from(value: String): Either[String, UserIri] =
-    if (value.isEmpty) Left(UserErrorMessages.UserIriMissing)
-    else {
-      val isUuid: Boolean = UuidUtil.hasValidLength(value.split("/").last)
-
-      if (!isUserIri(value))
-        Left(UserErrorMessages.UserIriInvalid(value))
-      else if (isUuid && !UuidUtil.hasSupportedVersion(value))
-        Left(IriErrorMessages.UuidVersionInvalid)
-      else
-        validateAndEscapeUserIri(value).toRight(UserErrorMessages.UserIriInvalid(value)).map(UserIri(_))
-    }
+final case class UserIri private (value: String) extends AnyVal with StringValue
+
+object UserIri extends StringValueCompanion[UserIri] {
 
-  def unsafeFrom(value: String): UserIri =
-    UserIri.from(value).fold(e => throw new IllegalArgumentException(e), identity)
+  implicit class UserIriOps(val userIri: UserIri) {
+    def isBuiltInUser: Boolean = builtInIris.contains(userIri.value)
+    def isRegularUser: Boolean = !isBuiltInUser
+  }
 
+  def makeNew: UserIri = UserIri.unsafeFrom(s"http://rdfh.ch/users/${UuidUtil.makeRandomBase64EncodedUuid}")
+
+  /**
+   * Explanation of the user IRI regex:
+   *
+   * `^` Asserts the start of the string.
+   *
+   * `http://rdfh\.ch/users/`: Matches the specified prefix.
+   *
+   * `[a-zA-Z0-9_-]{4,36}`: Matches any alphanumeric character, hyphen, or underscore between 4 and 36 times.
+   *
+   * `$`: Asserts the end of the string.
+   */
+  private val userIriRegEx = """^http://rdfh\.ch/users/[a-zA-Z0-9_-]{4,36}$""".r
+
+  private val builtInIris = Seq(
+    OntologyConstants.KnoraAdmin.SystemUser,
+    OntologyConstants.KnoraAdmin.AnonymousUser,
+    OntologyConstants.KnoraAdmin.SystemAdmin
+  )
+
+  private def isValid(iri: String) =
+    builtInIris.contains(iri) || (Iri.isIri(iri) && userIriRegEx.matches(iri))
```

Comment:
```suggestion
    builtInIris.contains(iri) && userIriRegEx.matches(iri))
```
IMO this redundant and `(Iri.isIri(iri)` is already covered by `userIriRegEx`.

Reply:
Probably you are right. Still, I would like to keep it as an additional check.

Also your suggestion is wrong,this would be the right predicate:
```scala
    builtInIris.contains(iri) || userIriRegEx.matches(iri)
```

Reply:
Yes, removed wrong operator.